lib/passageidentity/user_api.rb

```diff
@@ -108,105 +86,71 @@ def deactivate(user_id:)
         response = @user_client.deactivate_user(@app_id, user_id, @req_opts)
         response.user
       rescue Faraday::Error => e
-        if e.is_a? Faraday::ResourceNotFound
-          raise PassageError.new(
-            message: "Passage User with ID \"#{user_id}\" does not exist",
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        else
-          raise PassageError.new(
-            message: 'failed to deactivate Passage User.',
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        end
+        raise PassageError.new(
+          status_code: e.response[:status],
+          body: e.response[:body]
+        )
       end
     end
 
     def update(user_id:, email: '', phone: '', user_metadata: {})
+      warn '[DEPRECATED] the `update` method parameters will change to `user_id: string, ' \
+           'options: UpdateUserArgs`. Parameters will change on or after 2025-1.'
+
       user_exists?(user_id)
 
       updates = {}
       updates['email'] = email unless email.empty?
       updates['phone'] = phone unless phone.empty?
       updates['user_metadata'] = user_metadata unless user_metadata.empty?
-      begin
-        response = @user_client.update_user(@app_id, user_id, updates, @req_opts)
-        response.user
-      rescue Faraday::Error => e
-        if e.is_a? Faraday::ResourceNotFound
-          raise PassageError.new(
-            message: "Passage User with ID \"#{user_id}\" does not exist",
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        else
-          raise PassageError.new(
-            'failed to update Passage User',
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        end
-      end
+
+      update_v2(user_id: user_id, options: updates)
     end
 
     def create(email: '', phone: '', user_metadata: {})
+      warn '[DEPRECATED] the `create` method parameters will change to `args: CreateUserArgs`.' \
+      'Parameters will change on or after 2025-1.'
+
       create = {}
       create['email'] = email unless email.empty?
       create['phone'] = phone unless phone.empty?
       create['user_metadata'] = user_metadata unless user_metadata.empty?
-      begin
-        response = @user_client.create_user(@app_id, create, @req_opts)
-        response.user
-      rescue Faraday::Error => e
-        raise PassageError.new(
-          'failed to create Passage User',
-          status_code: e.response[:status],
-          body: e.response[:body]
-        )
-      end
+
+      create_v2(args: create)
     end
 
     def delete(user_id:)
       user_exists?(user_id)
 
       begin
         @user_client.delete_user(@app_id, user_id, @req_opts)
-        true
```

Comment:
this needs to still return a boolean, otherwise it's a breaking change since this method already existed

Reply:
It still returns the result of the successful `@user_client.delete_user(@app_id, user_id, @req_opts)`. Does the `/apps/{app_id}/users/{user_id}` delete not return true?